app/models/slide.rb

```diff
@@ -107,8 +110,38 @@ def take_screenshot
     end
   end
 
+  def file_size
+    if !@skip_file_validation
+      if foreground_type.present? && foreground_type != 'none'
+        self.send("#{foreground_type}_file", 'foreground' , {width: 912, height: 1080})
+      end
+
+      if background_type.present? && background_type != 'none'
+        self.send("#{background_type}_file", 'background', {width: 1920, height: 1080})
+      end
+    else
+      @skip_file_validation = false
+    end
+  end
+
   private
     def touch_signs
       signs.update_all(updated_at: Time.now)
     end
-end
+
+    def video_file(type, opts = nil)
```

Comment:
I totally agree that the method names blow here. I was coming up with naming quickly before I had to leave for my flight, but the thing about splitting into a foreground and a background method was a fork in the road in some ways. It would either be splitting methods on the file types (video vs image) or splitting methods on the placement type (foreground vs. background) and I figured the former was the better way to go.